### ISSUE TITLE
Titleize on SafeBuffer with non-alphanumeric values results in nil.capitalize exception

### ISSUE THREAD

**Issue by KieranP** [rails/rails#21808]
The following code is failing in Rails 4.2.4:

```
ActiveSupport::SafeBuffer.new("test_test_again").titleize
ActiveSupport::SafeBuffer.new("test test again").titleize
```

while these work fine:

```
"test_test_again".titleize
"test test again".titleize
ActiveSupport::SafeBuffer.new("test").titleize
ActiveSupport::SafeBuffer.new("test_test_again").capitalize
ActiveSupport::SafeBuffer.new("test test again").capitalize
```

Seems the regular expression in titleize inflector doesn't like non-alphanumeric values when it is within a safe buffer. When in a normal string though, this works without issues.

All examples worked in Rails 3.2.

A very weird issue.

Edit: Appears to have been picked up before and fixed in this pull request:
https://github.com/rails/rails/pull/19428

**Comment by Aeyrix** [comment]
A fix is proposed in #19428, which appears to have dropped off the radar as of March.

**Comment by rafaelfranca** [comment]
Fixed by #19428
